cpp/src/arrow/compute/CMakeLists.txt

```
@@ -114,6 +114,46 @@ function(ADD_ARROW_COMPUTE_TEST REL_TEST_NAME)
                  ${ARG_UNPARSED_ARGUMENTS})
 endfunction()
 
+# This function is used to add a custom main to the benchmarks in order
+# to initialize the compute kernels registry before running them.
+# This is necessary for benchmarks that use compute kernels that are not
+# part of libarrow.
+# It will also link the compute libraries to the benchmark target.
+function(ADD_ARROW_COMPUTE_BENCHMARK REL_TEST_NAME)
```

Could you use lower case for newly added function names?

```suggestion
function(add_arrow_compute_benchmark REL_TEST_NAME)
```

(And I want to use `arrow_` prefix for our CMake functions/macros...)

---

do you mean you want to start using `arrow_add_compute_benchmark` instead?

---

Yes.

See also: https://github.com/apache/arrow/pull/39824#discussion_r1505939524

---

It makes snse.